File: socoolbus/lib/models/expense.dart
class Expense {
  final String expenseId;
  final int amount;
  final String type;
  final DateTime date;
  final String description;

  const Expense(
      {required this.expenseId,
      required this.amount,
      required this.type,
      required this.date,
      required this.description});

  factory Expense.fromJson(Map<dynamic, dynamic> json) {
    return Expense(
      expenseId: json['_id'],
      amount: json['amount'],
      type: json['expenseType'],
      date: DateTime.parse(json['date']),
      description: json['description'],
    );
  }
}

File: server/models/expense.js
const mongoose = require("mongoose");

const ExpenseSchema = new mongoose.Schema({
	amount: {
		type: Number,
		required: true,
	},
	expenseType: {
		type: String,
		enum: ["Yakıt", "Yedek Parça", "Muhtelif", "Şahsi"],
		required: true,
	},
	date: {
		type: Date,
		required: true
	},
	description: {
		type: String,
	},
	phone: {
		//to understand which drivers' expenses they are
		type: String,
		required: true
	},
});

module.exports = Expense = mongoose.model("Expense", ExpenseSchema);
